src/main/java/com/cloudant/client/api/CloudantClient.java

```diff
@@ -247,7 +271,8 @@ public CloudantClient(String account, String authCookie, ConnectOptions connectO
      */
     public ApiKey generateApiKey() {
         URI uri = buildUri(getBaseUri()).path("_api/v2/api_keys").build();
-        return client.executeRequest(createPost(uri, "", ""), ApiKey.class);
+        HttpConnection connection = Http.POST(uri, "application/json");

```

Comment:
We should keep the usage of `HttpConnection` in as few classes as possible. We will need to set interceptors on every created `HttpConnection` and also potentially other parameters (e.g. `Proxy`). As such it would be beneficial to have all usages in one place.
So a better solution here might be to call a `post(uri)` method (e.g. the one on `CouchDatabaseBase`).